docs/architecture/adr-018-improve-subspace-treasury-account.md

```diff
@@ -0,0 +1,74 @@
+# ADR 018: Improve subspace treasury account
+
+## Changelog
+- Jan 4th, 2023: First draft;
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+This ADR introduces a new treasury account structure, which guarantees the account fully controlled by the subspace.
+
+## Context
+
+Currently, each subspace's treasury is a third-party accounts assigned by the subspace's manager(s). The subspace itself has no control over its treasury, which can lead to the following issues:
+1. subspace managers can assign a wealthy account not controlled by them as their treasury in order to scam users.;
+2. if we implement a feature that allows managers to spend money from the treasury in the future, they could assign any other account in order to steal funds.
+
+## Decision
+
+To address the issues mentioned above, we propose implementing a new treasury account structure. The new treasury address will be generated using `authtypes.NewModuleAddress` from its subspace id when the subspace is created, and will not be able to be edited thereafter. This will ensure that the subspace's treasury is fully controlled by the subspace itself, rather than being managed by external third-party accounts.
+
+Additionally, we will introduce a new permission and method to allow authorized users to spend funds from the treasury. 
+
+## `Msg` Service
+
+To prevent the treasury account from being edited later, we will remove the `treasury` field from the current `MsgCreateSubspace` and `MsgEditSubspace` messages.
+
+Additionally, we will implement a new operation to allow authorized users to spend funds from the treasury account as follows:
+
+```protobuf
+service Msg {
+    // SpendTreasury allows users who have the permission to transfer tokens out of the treasury
+    rpc SpendTreasury(MsgSpendTreasury) returns (MsgSpendTreasuryResponse);
+}
```

Comment:
I think the treasury should be only used as a place to store funds so having a method `SpendTreasury` is enough.

Reply:
I honestly don't believe this is enought. What we could do, instead, is levereging the `x/authz` module allowing subspace managers to create authorizations on behalf of this account. This would allow a lot of other use cases to be implemented. 

**Example** 
Suppose that a manager wants to use the treasury account to stake the tokens that are sent to it, in order to prove the users that the subspace owners are not going to dump on them. In this case, the subspace owner (or anyone with the permissions) could create a `StakeAuthorization` on behalf of the treasury, with the grantee being the user that wants to stake the tokens.   

This could be done by implementing one method that is `GrantAuthorization` to grant an authorization on behalf of the treasury. It would also cover the spending of the funds thanks to the `SpendAuthorization`

Reply:
You are right! Considering that treasury account might have cw20 tokens in the future, we should allow operators to execute smart contract, too.

Reply:
@RiccardoM Updated